### PR TITLE
Add Projectile Test Plugin

### PR DESCRIPTION
**Pull request by ryantheleach**
I made a test plugin whilst testing what Projectile events and causes were currently being thrown, Figured I'd chuck it your way if you were interested.

Issue: https://github.com/SpongePowered/SpongeCommon/issues/1397

### CODE REVIEW

File: testplugins/src/main/java/org/spongepowered/test/ProjectileTest.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+ */
+@Plugin(id = "projectiletest", name = "Projectile Test", description = "A plugin to test projectiles")
+public class ProjectileTest {
+
+    private static Text ENABLE = Text.of("enable");
+    private static Text unknown = Text.of("Unknown");
+
+    private boolean logging = false;
+
+    @Listener public void onGameStartingServer(GameStartingServerEvent event) {
```
Comment (gabizou) [inline testplugins/src/main/java/org/spongepowered/test/ProjectileTest.java]: Annotations on a new line.

File: testplugins/src/main/java/org/spongepowered/test/ProjectileTest.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+                this,
+                CommandSpec.builder()
+                        .arguments(GenericArguments.bool(ENABLE))
+                        .description(Text.of("Enable logging of projectiles"))
+                        .executor((src, args) -> {
+                            this.logging = args.<Boolean>getOne(ENABLE).get();
+                            return CommandResult.success();
+                        })
+                        .build()
+                , "logProjectiles");
```
Comment (gabizou) [inline testplugins/src/main/java/org/spongepowered/test/ProjectileTest.java]: If you're going to have things on separate lines, the comma is needed on the previous line and the parentheses on a new line.

File: testplugins/src/main/java/org/spongepowered/test/ProjectileTest.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        this.title("UseItemStack");
+        this.broadcast("Cause", event.getCause());
+        this.broadcast("Stack", event.getItemStackInUse());
+    }
+
+    @Listener public void onLaunchProjectile(LaunchProjectileEvent event) {
+        if (!logging) {
+            return;
+        }
+        title("LaunchProjectile");
```
Comment (gabizou) [inline testplugins/src/main/java/org/spongepowered/test/ProjectileTest.java]: Are you using `this` to qualify methods or not? Keep it uniform please ;)

**Review by gabizou — CHANGES_REQUESTED** [review]
Just style changes then ready for merging 👍 Good addition for the test plugin suite.

**Comment by ryantheleach** [comment]
I realized afterwards it would simplify the listening code if I just made the command register the listeners, I had to put it into an inner class, otherwise the listeners would have been enabled by default and needed to be removed. As so far, all test plugins have been single files, so that's why I decided to opt for an inner class as opposed to a normal class.

**Review by parlough — APPROVED** [review]
Seems like a useful test to add to me.

**Comment by parlough** [comment]
@ryantheleach You have this in your to-do list, is there anything else you'd like to do with this or is it good to be merged in?

**Comment by ryantheleach** [comment]
@Meronat I've just come back from holiday, and will start working on my TODO lists, on days that my bedroom is cooler then the surface of the sun.
  
Main things to address were gab's comments

**Pull request merged**